salt/hosts-minion/init.sls

```diff
@@ -0,0 +1,14 @@
+{%- for minion_id, addrlist in salt['mine.get']('roles:kube-master', 'network.ip_addrs', 'grain').items() %}
+{{ minion_id }}-host-entry:
+  host.present:
+{% if addrlist is string %}
+    - ip: {{ addrlist }}
+{% else %}
+    - ip: {{ addrlist|first }}
+{% endif %}
+    - names:
+      - api
+      - api.{{ pillar['internal_infra_domain'] }}
+      - {{ minion_id }}
+      - {{ minion_id }}.{{ pillar['internal_infra_domain'] }}
+{% endfor %}
```

Comment:
When we deploy HAProxy to all the nodes for load balancing / HA for the kube-api we will change `api.infra.caasp.local` to `127.0.0.1` and then use the IDs in the HAProxy upstream config, so keeping these entries will avoid us having to make more changes at a later date.